### PR TITLE
Clarify documentation about adding a new node type

### PR DESCRIPTION
**Pull request by Januszpl**
Related to #69 

First commit is quite clear - moving code to external file to improve extendability via js mixins if someone would need to add/change something here

Second commit - possible way of loading additional components for new node types. We just need backend dev to return array in nodes.phtml (as suggested by @Koc)

### CODE REVIEW

**Review by Igloczek — COMMENTED** [review]
It looks okay, should do the job.

File: view/adminhtml/templates/menu/nodes.phtml [hunk 1]
```diff
@@ -4,6 +4,18 @@ $fieldData = [];
 foreach ($block->getNodeForms() as $form) {
     $fieldData[$form->getNodeType()] = $form->getJsonConfig();
 }
+
+// TODO: backend - create method to return array of nodes components
+// https://github.com/SnowdogApps/magento2-menu/issues/69#issuecomment-443500749
+$vueComponents = [
```
Comment (Koc) [inline view/adminhtml/templates/menu/nodes.phtml:10]: We can create some `Provider` service with default node types and end-users of the module can create plugin/interceptor for this `Provider` to merge own additional node types

File: Model/VueProvider.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * @return array
+     */
+    public function getComponents(): array
+    {
+        $data = [];
+        foreach ($this->providers as $vueComponent) {
+            $data[] = sprintf('vue!Snowdog_Menu/vue/menu-type/%s', $vueComponent);
```
[the remark this answers is not in the document]
Comment (kiber) [reply Model/VueProvider.php]: Done

**Review by jochhop — APPROVED** [review]

**Review by KamilBalwierz — APPROVED** [review]

**Review by Igloczek — APPROVED** [review]
Made some minor changes, but it looks good from frontend perspective.

**Pull request merged**